storage/badger/cleaner.go

```diff
@@ -10,78 +10,95 @@ import (
 	"github.com/rs/zerolog"
 
 	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/component"
+	"github.com/onflow/flow-go/module/irrecoverable"
 )
 
+// Cleaner uses component.ComponentManager to implement module.Startable and module.ReadyDoneAware
+// to run an internal goroutine which run badger value log garbage collection on timely basis.
 type Cleaner struct {
-	log     zerolog.Logger
-	db      *badger.DB
-	metrics module.CleanerMetrics
-	enabled bool
-	ratio   float64
-	freq    int
-	calls   int
+	*component.ComponentManager
+	log      zerolog.Logger
+	db       *badger.DB
+	metrics  module.CleanerMetrics
+	ratio    float64
+	interval time.Duration
 }
 
-// NewCleaner returns a cleaner that runs the badger value log garbage collection once every `frequency` calls
-// if a frequency of zero is passed in, we will not run the GC at all
-func NewCleaner(log zerolog.Logger, db *badger.DB, metrics module.CleanerMetrics, frequency int) *Cleaner {
+var _ component.Component = (*Cleaner)(nil)
+
+// NewCleaner returns a cleaner that runs the badger value log garbage collection once every `interval` duration
+// if an interval of zero is passed in, we will not run the GC at all.
+func NewCleaner(log zerolog.Logger, db *badger.DB, metrics module.CleanerMetrics, interval time.Duration) *Cleaner {
 	// NOTE: we run garbage collection frequently at points in our business
 	// logic where we are likely to have a small breather in activity; it thus
 	// makes sense to run garbage collection often, with a smaller ratio, rather
 	// than running it rarely and having big rewrites at once
 	c := &Cleaner{
-		log:     log.With().Str("component", "cleaner").Logger(),
-		db:      db,
-		metrics: metrics,
-		ratio:   0.2,
-		freq:    frequency,
-		enabled: frequency > 0, // Disable if passed in 0 as frequency
+		log:      log.With().Str("component", "cleaner").Logger(),
+		db:       db,
+		metrics:  metrics,
+		ratio:    0.2,
+		interval: interval,
 	}
-	// we don't want the entire network to run GC at the same time, so
-	// distribute evenly over time
-	if c.enabled {
-		c.calls = rand.Intn(c.freq)
+
+	cmBuilder := component.NewComponentManagerBuilder()
+
+	// Disable if passed in 0 as interval
+	if c.interval > 0 {
+		cmBuilder.AddWorker(c.gcWorkerRoutine)
 	}
+
+	c.ComponentManager = cmBuilder.Build()
```

Comment:
When interval is 0, this builds an empty `ComponentManager` which is effectively a `module.NoopComponent`, but with added complexity. Instead, what do you think about
1. instead of embedding `*component.ComponentManager`, use `component.Component`. They are effectively the same minus `ShutdownSignal()`
2. when interval is 0, set `c.Component = &module.NoopComponent{}`

e.g.
```suggestion
	// Disable if passed in 0 as interval
	if c.interval == 0 {
		c.Component = &module.NoopComponent{}
		return c
	}

	c.Component := component.NewComponentManagerBuilder().
		AddWorker(c.gcWorkerRoutine).
		Build()
```

Reply:
Yeah, we can do this

Reply:
the only thing is that I can't find `NoopComponent`